### PR TITLE
refactor(localisation): restructure the localisation files

### LINKED ISSUE #17: Rework translation files
Right now the translation files are quite messy and a bit messy, so these could be easily reworked.

### PR DESCRIPTION
**Pull request by CodeNamedRobin**
# Description
Restructures the localisation files into multiple, smaller bits. This makes it easier to manage all different translations.

## Related issues/external references
Closes #17.

## Types of changes

- Style _(Change that do not affect the functionality of the code)_

### CODE REVIEW

**Comment by SuperVK** [comment]
image
Is there no way to do this without the hacky way? Like properly have it at the first level

**Comment by CodeNamedRobin** [comment]
> image Is there no way to do this without the hacky way? Like properly have it at the first level

Nope, there is no way to do this with the current ESLint plugin. And I do want to keep that plugin as it enforces that localisation actually works.

File: apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue [hunk 1]
```diff
@@ -1,22 +1,21 @@
 <template>
   <div class="page-container">
-    <div class="page-title">{{ `${currentUser ? currentUser.firstName : ''}'s profile` }}</div>
-
+    <div class="page-title">
+      {{ t('modules.admin.singleUser.profile', {user: currentUser ? currentUser.firstName : ''}) }}
```
Comment (Gijsdeman) [inline apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue:4]:
```suggestion
      {{ t('modules.admin.singleUser.profile', {user: currentUser ?? currentUser.firstName}) }}
```

File: apps/dashboard/src/locales/nl/components/forms.json
Comment (Gijsdeman) [inline apps/dashboard/src/locales/nl/components/forms.json:1]: Is there a reason to keep the file even though it does not contain any translations?

File: apps/dashboard/src/locales/en/components/forms.json
Comment (Gijsdeman) [inline apps/dashboard/src/locales/en/components/forms.json:1]: Is there a reason to keep empty translation file?

File: .eslintrc.cjs [hunk 2]
```diff
@@ -19,8 +21,8 @@ module.exports = {
         'vue/block-lang': ['error', { 'script': { 'lang': 'ts' } }],
         '@intlify/vue-i18n/no-raw-text': 'error',
         '@intlify/vue-i18n/no-missing-keys': 'error',
-        '@intlify/vue-i18n/no-missing-keys-in-other-locales': 'error',
-        '@intlify/vue-i18n/no-unused-keys': ['error', { 'enableFix': true, 'extensions': ['.js', '.vue', '.ts'] }],
+        '@intlify/vue-i18n/no-missing-keys-in-other-locales': ['error', ],
```
Comment (Gijsdeman) [inline .eslintrc.cjs]:
```suggestion
        '@intlify/vue-i18n/no-missing-keys-in-other-locales': ['error'],
```

or even 

```suggestion
        '@intlify/vue-i18n/no-missing-keys-in-other-locales': ['error', {}],
```

File: apps/dashboard/src/components/container/ContainersCard.vue [hunk 3]
```diff
@@ -1,12 +1,12 @@
...
+  <CardComponent :header="t('modules.seller.productContainers.containers.header')" class="p-0">
     <template #topAction>
       <div v-if="showCreate || associatedPos" class="flex justify-content-endg gap-2">
         <Button
             v-if="associatedPos"
             @click="openContainerAdd()"
-            outlined>{{ t('containersOverview.addExisting')}}</Button>
-        <Button @click="openContainerEdit()">{{t('app.Create')}}</Button>
+            outlined>{{ t('modules.seller.productContainers.containers.addExisting') }}</Button>
+        <Button @click="openContainerEdit()">{{t('common.create')}}</Button>
```
Comment (Gijsdeman) [inline apps/dashboard/src/components/container/ContainersCard.vue]: Maybe add something along the lines of [vue/mustache-interpolation-spacing](https://eslint.vuejs.org/rules/mustache-interpolation-spacing) to enforce consistent spacing for template syntax (most of the times you use single spacing around, in some rare cases you forget it).

File: apps/dashboard/src/utils/i18nUtils.ts [hunk 4]
```diff
@@ -9,28 +27,65 @@ const i18n = createI18n({
     legacy: false,
     globalInjection: true,
     messages: {
```
Comment (Gijsdeman) [inline apps/dashboard/src/utils/i18nUtils.ts:29]: Why are the `components` and `modules` fields present in the imported json files?

I think it my be clearer if you omit those fields from the imported json files (and thus move all fields up a level), and instead import the files as, for example, `footer_en_components`, and destruct those json files directly.

File: apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue (hunk 1 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/modules/admin/views/AdminSingleUserView.vue:4]: This does not work as it then interprets the type of currentUser as `never`

File: apps/dashboard/src/locales/nl/components/forms.json
Comment (CodeNamedRobin) [reply apps/dashboard/src/locales/nl/components/forms.json:1]: Not really, removed in latest commit.

File: apps/dashboard/src/locales/en/components/forms.json
Comment (CodeNamedRobin) [reply apps/dashboard/src/locales/en/components/forms.json:1]: Not really, removed in last commit.

File: .eslintrc.cjs (hunk 2 above)
Comment (CodeNamedRobin) [reply .eslintrc.cjs]: Resolved in latest commit.

File: apps/dashboard/src/components/container/ContainersCard.vue (hunk 3 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/components/container/ContainersCard.vue]: Added, is indeed better.

File: apps/dashboard/src/utils/i18nUtils.ts (hunk 4 above)
Comment (CodeNamedRobin) [reply apps/dashboard/src/utils/i18nUtils.ts:29]: I don't know exactly what you mean? If you mean the keys in the json files themselves, that is so the lintr recognises everything. Other setups don't work, I've tried.

**Review by Gijsdeman — APPROVED** [review]

**Comment by JustSamuel** [comment]
`yarn build` fails for me with 

```
error during build:
Could not resolve "../locales/en.json" from "src/utils/i18nUtils.ts"
```

**Pull request merged**